homeassistant/components/sensor/bom.py

```diff
@@ -66,25 +74,53 @@
     'wind_spd_kt': ['Wind Direction kt', 'kt']
 }
 
+
+def validate_station(station):
+    """Check that the station ID is well-formed."""
+    if station is None:
+        return
+    station = station.replace('.shtml', '')
+    if not re.fullmatch(r'ID[DNQTVW]60801\.\d\d\d\d\d', station):
```

Comment:
We won't autodetect such stations I think, so it's particularly important to allow manual config!  Fixed.